commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java

```diff
@@ -778,4 +778,14 @@ void testMath340() {
                                                    fractionA.getDenominator().multiply(fractionB.getDenominator()));
         Assertions.assertEquals(correctResult, errorResult);
     }
+
+    @Test
+    void testNumbers150() {
+        // zero to negative powers should throw an exception
+        Assertions.assertThrows(ArithmeticException.class, () -> BigFraction.ZERO.pow(-1));
+        Assertions.assertThrows(ArithmeticException.class, () -> BigFraction.ZERO.pow(Integer.MIN_VALUE));
+
+        // shall overflow
+        Assertions.assertThrows(ArithmeticException.class, () -> Fraction.of(2).pow(Integer.MIN_VALUE));
```

Comment:
I would add `Fraction.of(1, 2)` as well to test overflow in the numerator or denominator

Reply:
@aherbert done.